crates/miden-lib/src/transaction/script_builder.rs

```diff
@@ -0,0 +1,177 @@
+    /// transaction will expire.
+    expiration_delta: Option<u16>,
+    /// Indicates if the script should be compiled in debug mode.
+    in_debug_mode: bool,
+}
+
+impl TransactionScriptBuilder {
+    pub fn new(
+        account_interface: AccountInterface,
+        expiration_delta: Option<u16>,
+        in_debug_mode: bool,
+    ) -> Self {
+        Self {
+            account_interface,
+            expiration_delta,
+            in_debug_mode,
+        }
+    }
+
+    /// Builds a transaction script which sends the specified notes with the corresponding
+    /// authentication.
+    pub fn build_send_notes_script(
+        &self,
+        output_notes: &[PartialNote],
+    ) -> Result<TransactionScript, TransactionScriptBuilderError> {
+        let send_note_procedure = if self
+            .account_interface
+            .components()
+            .contains(&AccountComponentInterface::BasicFungibleFaucet)
+        {
+            AccountComponentInterface::BasicFungibleFaucet
+                .send_note_procedure(*self.account_interface.id(), output_notes)?
+        } else if self
+            .account_interface
+            .components()
+            .contains(&AccountComponentInterface::BasicWallet)
+        {
+            AccountComponentInterface::BasicWallet
+                .send_note_procedure(*self.account_interface.id(), output_notes)?
+        } else {
+            return Err(TransactionScriptBuilderError::UnsupportedAccount);
+        };
+
+        self.build_script_with_sections(&[send_note_procedure])
+    }
+
+    /// Builds a simple authentication script for the transaction that doesn't send any notes.
+    pub fn build_auth_script(&self) -> Result<TransactionScript, TransactionScriptBuilderError> {
+        self.build_script_with_sections(&[])
+    }
+
+    /// Builds a transaction script with the specified sections.
+    ///
+    /// The `sections` parameter is a vector of strings, where each string represents a distinct
+    /// part of the script body. The script includes, authentication, and expiration sections are
+    /// automatically added to the script.
+    fn build_script_with_sections(
+        &self,
+        sections: &[String],
+    ) -> Result<TransactionScript, TransactionScriptBuilderError> {
+        let script = format!(
+            "{} begin {} {} {} end",
+            self.script_includes()?,
+            self.script_expiration(),
+            sections.join(" "),
+            self.script_authentication()
+        );
+
+        let assembler = TransactionKernel::assembler().with_debug_mode(self.in_debug_mode);
+        let tx_script = TransactionScript::compile(script, vec![], assembler)
+            .map_err(TransactionScriptBuilderError::InvalidTransactionScript)?;
+
+        Ok(tx_script)
+    }
+
+    /// Returns a string with the needed include instructions for the script.
+    fn script_includes(&self) -> Result<String, TransactionScriptBuilderError> {
```

Comment:
Is the intention to produce a "nice" script in terms of readability? If not, could we not remove the include logic and use fully qualified names of the procedures instead? That would enable us to remove some code and keep it simpler.

Reply:
At some point we showed the scripts in the CLI but I agree we could probably remove this now and simplify it.